### PR TITLE
Outline modernizing tool shed repositories API.

### PR DESCRIPTION
**Pull request by jmchilton**
Builds on #14693 and migrates a simple endpoint to Pydandic models, a DI injected service layer, refactored utility code with decoded IDs and reduced dependency on ``app`` && ``trans``, and FastAPI controller.

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py [hunk 1]
```diff
@@ -474,3 +473,17 @@ def show(self, trans, id, **kwd):
...
+
+
+@router.cbv
+class FastAPIToolShedRepositories:
+    service: ToolShedRepositoriesService = depends(ToolShedRepositoriesService)
+
+    @router.get(
+        "/api/tool_shed_repositories/check_for_updates",
+        summary="Lists stored workflows viewable by the user.",
+        response_description="A list with summary stored workflow information per viewable entry.",
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py]: Looks like some leftovers ;)

File: lib/galaxy/schema/schema.py [hunk 2]
```diff
@@ -2194,6 +2194,25 @@ class GroupUserListModel(Model):
...
+    name: str = Field(name="Name", description="Name of repository")
+    owner: str = Field(name="Owner", description="Owner of repository")
+
+
+class ToolShedRepositoryChangeset(ToolShedRepository):
+    changeset_revision: str
+
+
+class CheckForUpdatesResponse(Model):
+    status: str = Field(title="Status", description="'ok' or 'error'")
```
Comment (davelopez) [inline lib/galaxy/schema/schema.py]: Is it worth creating an enum for the status?

File: lib/galaxy/webapps/galaxy/services/tool_shed_repositories.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from pydantic import BaseModel
+
+from galaxy.model.scoped_session import install_model_scoped_session
+from galaxy.tool_shed.util.repository_util import check_for_updates
+from galaxy.util.tool_shed.tool_shed_registry import Registry
+
+
+class CheckForUpdatesResponse(BaseModel):
+    status: str
+    message: str
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/services/tool_shed_repositories.py]: This one can be removed and imported from `schema.py` :)

**Review by davelopez — COMMENTED** [review]
Is it still in draft? Looks pretty solid to me. 

Just some minor comments below :)

File: lib/galaxy/schema/schema.py (hunk 2 above)
Comment (nsoranzo) [reply lib/galaxy/schema/schema.py]: In case, use [Literal](https://docs.python.org/3/library/typing.html#typing.Literal), not enums, please.

**Review by bgruening — APPROVED** [review]

**Pull request merged**